src/webhooks/pubsub/stalebot.ts

```diff
@@ -16,68 +16,96 @@ const staleStatusUpdater = async (
   await Promise.all(
     issues.map((issue) => {
       const isPullRequest = issue.pull_request ? true : false;
-      const issueHasStaleLabel = issue.labels.some(
-        (label) => label === STALE_LABEL || label.name === STALE_LABEL
-      );
-      if (issueHasStaleLabel) {
-        if (now.diff(issue.updated_at, 'days') >= DAYS_BEFORE_CLOSE) {
-          // Interestingly enough, this api works for both issues and pull requests
-          return org.api.issues.update({
+      if (now.diff(issue.updated_at, 'days') >= DAYS_BEFORE_STALE) {
+        return Promise.all([
+          org.api.issues.addLabels({
             owner: org.slug,
             repo: repo,
             issue_number: issue.number,
-            state: 'closed',
-          });
-        } else if (now.diff(issue.updated_at, 'days') < DAYS_BEFORE_CLOSE) {
-          return org.api.issues.removeLabel({
+            labels: [STALE_LABEL],
+          }),
+          org.api.issues.createComment({
             owner: org.slug,
             repo: repo,
             issue_number: issue.number,
-            name: STALE_LABEL,
-          });
-        }
-        return Promise.resolve();
-      } else {
-        if (now.diff(issue.updated_at, 'days') >= DAYS_BEFORE_STALE) {
-          return Promise.all([
-            org.api.issues.addLabels({
-              owner: org.slug,
-              repo: repo,
-              issue_number: issue.number,
-              labels: [STALE_LABEL],
-            }),
-            org.api.issues.createComment({
-              owner: org.slug,
-              repo: repo,
-              issue_number: issue.number,
-              body: `This ${
-                isPullRequest ? 'pull request' : 'issue'
-              } has gone three weeks without activity. In another week, I will close it.
+            body: `This ${
+              isPullRequest ? 'pull request' : 'issue'
+            } has gone three weeks without activity. In another week, I will close it.
```

Comment:
Do we have a test for this messaging? It seems like we only test issue closing behavior?

Reply:
Yes, we do :). Check https://github.com/getsentry/eng-pipes/blob/94fe7e20b581ccac63bea2a5dcfb1f0639870325/src/webhooks/pubsub/stalebot.test.ts#L62